docs/zh/blog/code/layotto-rpc/index.md

```diff
@@ -0,0 +1,921 @@
+    }
+    
+    // 6. 将结果转换为 rpc.RPCResponse 返回
+    rpcResp := &rpc.RPCResponse{
+        ContentType: string(httpResp.Header.ContentType()),
+        Data:        body,
+        Header:      map[string][]string{},
+    }
+    httpResp.Header.VisitAll(func(key, value []byte) {
+        rpcResp.Header[string(key)] = []string{string(value)}
+    })
+    return rpcResp, nil
+```
+
+### 0x09 Grpc Sever 处理数据帧返回给客户端
+
+Grpc 并没有直接写入数据到连接，而是用协程异步 loop 循环从一个缓存结构里面获取帧然后写回到客户端，具体流程如下。
+```go
+google.golang.org/grpc/internal/transport.NewServerTransport at http2_server.go
+func NewServerTransport(conn net.Conn, config *ServerConfig) (_ ServerTransport, err error) {
+    ...
+    // 协程异步loop循环
+    go func() {
+        t.loopy = newLoopyWriter(serverSide, t.framer, t.controlBuf, t.bdpEst)
+        t.loopy.ssGoAwayHandler = t.outgoingGoAwayHandler
+        if err := t.loopy.run(); err != nil {
+            if logger.V(logLevel) {
+                logger.Errorf("transport: loopyWriter.run returning. Err: %v", err)
+            }
+        }
+        t.conn.Close()
+        t.controlBuf.finish()
+        close(t.writerDone)
+    }()
+    go t.keepalive()
+    return t, nil
+}
+=>
+google.golang.org/grpc/internal/transport.(*loopyWriter).run at controlbuf.go
+=>
+google.golang.org/grpc/internal/transport.(*bufWriter).Flush at http_util.go
+=>
+mosn.io/mosn/pkg/filter/network/grpc.(*Connection).Write at conn.go
+=>
+mosn.io/mosn/pkg/network.(*connection).Write at connection.go
+=>
+mosn.io/mosn/pkg/network.(*connection).writeDirectly at connection.go
+=>
+mosn.io/mosn/pkg/network.(*connection).doWrite at connection.go
+```
+
+### 0x10 dubbo-go-sample client 接收响应
+
+接上述 0x01 发送数据之后会阻塞在 Client grpc 底层读IO中, Layotto经过上述一些列处理层层返回数据激活Client底层Read IO，具体流程如下。
+```go
+google.golang.org/grpc.(*ClientConn).Invoke at call.go
+=>
+google.golang.org/grpc.(*ClientConn).Invoke at call.go
+=>
+google.golang.org/grpc.(*clientStream).RecvMsg at stream.go
+=>
+google.golang.org/grpc.(*clientStream).withRetry at stream.go
+=>
+google.golang.org/grpc.(*csAttempt).recvMsg at stream.go
+=>
+google.golang.org/grpc.recvAndDecompress at rpc_util.go
+=>
+google.golang.org/grpc.recv at rpc_util.go
+=>
+google.golang.org/grpc.(*parser).recvMsg at rpc_util.go
+=>
+google.golang.org/grpc.(*csAttempt).recvMsg at stream.go
+func (p *parser) recvMsg(maxReceiveMessageSize int) (pf payloadFormat, msg []byte, err error) {
+    if _, err := p.r.Read(p.header[:]); err != nil {
+        return 0, nil, err
+    }
+    ...
+}
+```
+最终收到返回数据：
+{"jsonrpc":"2.0","id":9527,"result":{"id":"113","name":"Moorse","age":30,"time":703394193,"sex":"MAN"}}
+
+## 总结
+Layotto RPC 处理流程涉及 GRPC、Dapr、Mosn 等相关的知识，整体流程较长，不过单纯看 Layotto 针对 Mosn 抽象的轻量 RPC 框架还是比较清晰和简单的，与 Mosn 集成的方式也比较新颖，值得进一步研读。至此 Layotto RPC 请求处理就分析完了，时间有限，没有进行一些更全面和深入的剖析，如有纰漏之处，欢迎指正，联系方式：rayo.wangzl@gmail.com。另外在此也希望大家能踊跃参与源码分析和开源社区来，一起学习，共同进步。
```

Comment:
@MoonShining Haha, your code is highly appreciated :)
Could u help review this PR ?